contracts/src/v0.8/dev/ArbitrumValidator.sol

```diff
@@ -0,0 +1,140 @@
+   * @param gasCostL2 value to send to L2 to cover gas fee
+   * @param refundableAddress address where gas excess on L2 will be sent
+   */
+  constructor(
+    address aggregatorAddress,
+    address inboxAddress,
+    address flagAddress,
+    address gasConfigAccessController,
+    uint256 maxSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  ) {
+    require(flagAddress != address(0), "Invalid Flags contract address");
+    s_arbitrumInbox = ArbitrumInboxInterface(inboxAddress);
+    s_gasConfigAccessController = SimpleWriteAccessController(gasConfigAccessController);
+    s_flagsAddress = flagAddress;
+    _setGasConfiguration(maxSubmissionCost, maximumGasPrice, gasCostL2, refundableAddress);
+
+    this.addAccess(aggregatorAddress);
+  }
+  
+  fallback() external payable {}
+
+  function withdrawFunds() 
+    external 
+    onlyOwner() 
+  {
+    address payable to = payable(msg.sender);
+    to.transfer(address(this).balance);
+  }
+
+  function withdrawFundsTo(
+    address payable to
+  ) 
+    external
+    onlyOwner() 
+  {
+    to.transfer(address(this).balance);
+  }
+
+  function setGasConfiguration(
+    uint256 maxSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  )
+    external
+  {
+    require(s_gasConfigAccessController.hasAccess(msg.sender, msg.data), "Only gas configuration admin can call");
+    _setGasConfiguration(maxSubmissionCost, maximumGasPrice, gasCostL2, refundableAddress);
+  }
+
+  function validate(
+    uint256 /* previousRoundId */,
+    int256 /* previousAnswer */,
+    uint256 /* currentRoundId */,
+    int256 currentAnswer
+  ) 
+    external 
+    override
+    checkAccess() 
+    returns (bool) 
+  {
+    // Avoids resending to L2 the same tx on every call
+    if (s_lastChangedAnswer == currentAnswer) {
+      return true;
+    }
+    s_lastChangedAnswer = currentAnswer;
+    bytes memory data = currentAnswer == 1 ? abi.encodeWithSelector(RAISE_SELECTOR, s_arbitrumFlag) : abi.encodeWithSelector(LOWER_SELECTOR, s_arbitrumFlag);
+
+    s_arbitrumInbox.createRetryableTicket{value: s_gasConfig.gasCostL2}(
+      s_flagsAddress, 
+      0, // L2 call value
+      s_gasConfig.maximumSubmissionCost, // Max submission cost of sending data length
+      s_gasConfig.refundableAddress, // excessFeeRefundAddress
+      s_gasConfig.refundableAddress, // callValueRefundAddress
+      s_L2GasLimit,
+      s_gasConfig.maximumGasPrice, 
+      data
+    );
+    return true;
+  }
+  
+  event GasConfigurationSet(
+    uint256 maximumSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  );
+  
+  function _setGasConfiguration(
+    uint256 _maximumSubmissionCost,
+    uint32 _maximumGasPrice,
+    uint256 _gasCostL2,
+    address _refundableAddress
+  ) internal {
+    uint256 minGasCostValue = _maximumSubmissionCost + s_L2GasLimit * _maximumGasPrice;
```

Comment:
Let's use `SafeMath` for these operations.

Reply:
This is a 0.8 contract, no? If so, the solidity compiler gives you safe math "for free".

> Arithmetic operations revert on underflow and overflow. You can use unchecked { ... } to use the previous wrapping behaviour.


Source: https://docs.soliditylang.org/en/breaking/080-breaking-changes.html

Reply:
Ah yes, thx! I missed that this contract was migrated to v0.8. All good then!